gammapy/cube/make.py

```diff
@@ -257,3 +297,359 @@ def _check_selection(selection):
             raise ValueError("Selection not available: {!r}".format(name))
 
     return selection
+
+
+class MapMakerRing(MapMaker):
+    """Make maps from IACT observations.
+
+    The main motivation for this class in addition to the `MapMaker`
+    is to have the common image background estimation methods,
+    like `~gammapy.background.RingBackgroundEstimator`,
+    that work using on and off maps.
+
+    To ensure adequate statistics, only observations that are fully
+    contained within the reference geometry will be analysed
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.WcsGeom`
+        Reference image geometry
+    offset_max : `~astropy.coordinates.Angle`
+        Maximum offset angle
+    exclusion_mask : `~gammapy.maps.Map`
+        Exclusion mask
+    background_estimator : `~gammapy.background.RingBackgroundEstimator`
+        or `~gammapy.background.AdaptiveRingBackgroundEstimator`
+        Ring background estimator or something with an equivalent API.
+
+    """
+
+    def __init__(
+        self, geom, offset_max, exclusion_mask=None, background_estimator=None
+    ):
+        super(MapMakerRing, self).__init__(
+            geom=geom,
+            offset_max=offset_max,
+            exclusion_mask=exclusion_mask,
+            geom_true=None,
+        )
+        self.background_estimator = background_estimator
+
+    def _stack(self, maps_obs, obs_maker):
+        selection = ["counts", "exposure", "background"]
+        maps = self._get_empty_maps(selection)
+        for name in selection:
+            data = maps_obs[name].quantity.to_value(maps[name].unit)
+            if name == "exposure":
+                maps[name].fill_by_coord(obs_maker.coords_etrue, data)
+            else:
+                maps[name].fill_by_coord(obs_maker.coords, data)
+        return maps
+
+    def _run(self, observations, sum_over_axis=False, spectrum=None, keepdims=False):
+        """
+        Parameters
+        --------------
+        observations : `~gammapy.data.Observations`
+            Observations to process
+
+        Returns
+        -----------
+        maps: list of dict of maps. Each observation will have the following maps
+            counts: The counts map
+            background_irf: The template background map from the IRF
+            exposure: The uncorrelated on exposure map
+            exposure_off: The off exposure map convolved with the ring
+            alpha: The alpha map (1/exposure map)
+            off: The off map
+            background_ring: The ring background map ( = alpha * off)
+            exclusion: The exclusion mask
+
+        """
+
+        map_list = []
```

Comment:
This is not clear to me (also see my comment above). Does` _run()` return the maps per observations or the stacked ones?